pkg/kubelet/status/status_manager.go

```diff
@@ -483,6 +483,10 @@ func (m *manager) TerminatePod(pod *v1.Pod) {
 // hasPodInitialized returns true if the pod has no evidence of ever starting a regular container, which
 // implies those containers should not be transitioned to terminated status.
 func hasPodInitialized(pod *v1.Pod) bool {
+	// a pod in the running phase is always initialized by definition
+	if pod.Status.Phase == v1.PodRunning {
+		return true
+	}
```

Comment:
The logic below needs to change I think - if the last N init containers are RestartAlways and are not terminated but have passed their startup probe, then the pod is initailized?

Need to discuss this function a bit more.